pkgs/development/libraries/bencode/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, catch2
+, cmake
+, expected-lite
+, fmt
+, gsl-lite
+, ninja
+}:
+
+stdenv.mkDerivation rec {
+  pname = "bencode";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "fbdtemme";
+    repo = "bencode";
+    rev = "v${version}";
+    hash = "sha256-ePbQ6oTumQ+T5TOb+ReaOYa8Q3lvcMBZ6jTMoquTngg=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  buildInputs = [
+    catch2
+    expected-lite
+    fmt
+    gsl-lite
+  ];
+
+  postPatch = ''
+    # Disable a test that requires an internet connection.
+    substituteInPlace tests/CMakeLists.txt \
+      --replace "add_subdirectory(cmake_fetch_content)" ""
+  '';
+
+  doCheck = true;
+
+  postInstall = ''
+    rm -rf $out/lib64
```

Comment:
Is there anything useful on this directory, or it is merely created because the upstream developer didn't see it on the cmakefiles?

Reply:
I guess I just forgot to remove this when I was copying the derivation from somewhere.

Reply:
I am a bit concerned about this, because AFAIK nixpkgs does not split lib and lib64.

Reply:
It does not split, correct. I think the intention there was to remove `lib64` for that specific reason, and because the `lib` target is already in 64 bits.

Reply:
My bad, it does actually [break](https://logs.nix.ci/?key=nixos/nixpkgs.199662&attempt_id=1452624c-da55-4657-8555-306299acdc82) but only when the check phase is enabled.

Reply:
We need to delete `lib64` because it's created by the CMake installation target (although, I don't know exactly where). Here's the file structure:
```
bencode> |-- lib
bencode> |   `-- cmake
bencode> |       `-- bencode
bencode> |           |-- bencode-config-version.cmake
bencode> |           |-- bencode-config.cmake
bencode> |           `-- bencode-targets.cmake
bencode> `-- lib64
bencode>     `-- cmake
bencode>         `-- bencode
bencode>             |-- bencode-config-version.cmake
bencode>             |-- bencode-config.cmake
bencode>             `-- bencode-targets.cmake
```
All files in there are identical.

Reply:
OK then.